graph_connect.go

```diff
@@ -205,16 +193,18 @@ func attachArrayPort(port reflect.Value, key int, dir reflect.ChanDir, ch reflec
 		return ch, err
 	}
 
-	if port.IsNil() {
-		m := reflect.MakeSlice(port.Type(), 0, 32)
-		port.Set(m)
-	}
+	const scalingFactor = 2
 
-	if port.Cap() <= key {
-		port.SetCap(2 * key)
```

Comment:
Note that this is [guaranteed to panic](https://golang.org/src/reflect/value.go#L1671). To "extend" the underlying array, you have to reallocate the slice and copy the data.